pyproject.toml

```
@@ -16,7 +16,7 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.7"
-graphene = ">=3.0b6"
+graphene = "~=3.1"
```

I think the changes are possible without dropping support for graphene 3.0, GQLFormattedError should already exist in the former GQL-Core release - will check tomorrow.

---

It looks like `GQLFormattedError` was introduced in graphgl-core 3.2.0 in the same commit https://github.com/graphql-python/graphql-core/commit/09ff14f68bdce1e9cd71decc871fd38274b01971 (from diff https://github.com/graphql-python/graphql-core/compare/v3.1.7...v3.2.0)

Since graphene 3.1 is the first release to (properly) support graphgl-core 3.2.0, I think this lower pin makes sense. Previous versions of this lib [can](https://github.com/graphql-python/graphene/issues/1406#issuecomment-1087509664) still be used with graphql-core<3.2, risking [dephell](https://medium.com/knerd/the-nine-circles-of-python-dependency-hell-481d53e3e025) with the earlier versions of graphene/gql/graphql-core/graphql-relay.

---

from a closer look: graphene 3.1 still allows graphql-core 3.1, whereas this PR drops support. so either this PR should introduce compat import statements, or this PR should drop support:
```suggestion
graphene = "~=3.1"
graphql-core = "~=3.2"
```

what do you think?

---

Your suggestion seems to be necessary to prevent compatibility issues.  Whether or not to introduce compatibility import statements is a question of scope. This is a quick fix, compatibility imports would increase complexity. 
The cleaner way to go would be by supporting all core versions that graphene supports, aligning with the main library.

---

towards preventing future compatibility issues: https://github.com/ddelange/starlette-graphene3/pull/2

I think it's one step forward, plz let me know what you think :)